orderer/consensus/follower/follower_chain.go

```diff
@@ -199,31 +235,69 @@ func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
 	return clusterRelation, status
 }
 
+func (c *Chain) IsRunning() bool {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started {
+		select {
+		case <-c.doneChan:
+			return false
+		default:
+			return true
+		}
+	}
+
+	return false
+}
+
 func (c *Chain) run() {
 	c.logger.Debug("The follower.Chain puller goroutine is starting")
-	pullRetryTicker := time.NewTicker(500 * time.Millisecond) // TODO move constant to options
 
 	err := c.pull()
 
 runLoop:
 	for err != nil {
-		c.logger.Debugf("Pull failed, going to try again in %v; error: %s", pullRetryTicker, err)
+		c.logger.Debugf("Pull failed, going to try again in %v; error: %s", c.retryInterval, err)
 
 		select {
 		case <-c.stopChan:
 			c.logger.Debug("Received a stop signal")
 			break runLoop
 
-		case <-pullRetryTicker.C:
+		case <-time.After(c.retryInterval):
```

Comment:
why the change? the previous version is more testable no?

Reply:
The change is in order to be able to reset the timeout after a good block is received and not get stuck in a long retry interval after a sequence of failures.

Reply:
Can we at least extract the time.After function to the options so it'll be injectable and more testable?

Reply:
Yes, I'll do that, and add a unit test as well.

Reply:
Done